include/openthread/platform/radio.h

```diff
@@ -907,19 +907,6 @@ otError otPlatRadioReceiveAt(otInstance *aInstance, uint8_t aChannel, uint32_t a
  */
 extern void otPlatRadioReceiveDone(otInstance *aInstance, otRadioFrame *aFrame, otError aError);
 
-/**
- * The radio driver calls this function to notify OpenThread diagnostics module of a received frame.
- *
- * Is used when diagnostics is enabled.
- *
- * @param[in]  aInstance The OpenThread instance structure.
- * @param[in]  aFrame    A pointer to the received frame or NULL if the receive operation failed.
- * @param[in]  aError    OT_ERROR_NONE when successfully received a frame,
- *                       OT_ERROR_ABORT when reception was aborted and a frame was not received,
- *                       OT_ERROR_NO_BUFS when a frame could not be received due to lack of rx buffer space.
- */
-extern void otPlatDiagRadioReceiveDone(otInstance *aInstance, otRadioFrame *aFrame, otError aError);
```

Comment:
While I like the idea of removing these callbacks (as it would help simplify the code), I think it would be good to check the reason they were originally added.

Why did we have separate callbacks specifically for `Diag TxDone` and `RxDone`? I don't recall the history (though I was not really involved in the diagnostics-related topics). Does anyone happen to know?

Reply:
Here is the original [PR343](https://github.com/openthread/openthread/pull/343/files#diff-2714f9d39ad5f34b63c1869905a55384e1b522d9b848836a924a19cb1dbffc29R372) that adds these two APIs. These two APIs were added in the very early days of OpenThread. At that time, there are no `SubMac` and `Radio` modules in OpenThread. Both the Mac module and the diag module directly call the `otPlatRadioXxx` APIs, and they are two independent modules. In that case, the best way to send the same event to different modules is to use different event functions for different modules. Now, we have the Radio module, there is no need to use different event functions.